curl-helper.c

```diff
@@ -5377,6 +5377,100 @@ value caml_curl_check_enums(value v_unit)
   CAMLreturn(v_r);
 }
 
+/* WebSocket flags mapping: same order as OCaml variant */
+static const long wsFlags[] = {
+  CURLWS_TEXT,    /* 0 */
+  CURLWS_BINARY,  /* 1 */
+  CURLWS_CONT,    /* 2 */
+  CURLWS_CLOSE,   /* 3 */
+  CURLWS_PING,    /* 4 */
+  CURLWS_PONG,    /* 5 */
+  CURLWS_OFFSET   /* 6 */
+};
+
+static value curlWSFlag_list_of_int(int flags)
+{
+  CAMLparam0();
+  CAMLlocal1(result);
+
+  result = Val_emptylist;
+
+  for (int i = 0; i < (int)(sizeof(wsFlags) / sizeof(wsFlags[0])); i++) {
+    if (flags & wsFlags[i]) {
+      result = Val_cons(result, Val_int(i));
+    }
+  }
+
+  CAMLreturn(result);
+}
+
+
+static int curlWSFlag_list_to_int(value flag_list)
+{
+  CAMLparam1(flag_list);
+  long flags = convert_bit_list(wsFlags, sizeof(wsFlags) / sizeof(wsFlags[0]), flag_list);
+
+  CAMLreturn((int)flags);
```

Comment:
iirc correct way is CAMLreturnT? but actually can skip CAMLparam and CAMLreturn altogether here because there are no ocaml heap allocations in this func